src/main/kotlin/no/nav/familie/pdf/pdf/TabellUtils.kt

```diff
@@ -11,68 +11,73 @@ import com.itextpdf.layout.properties.UnitValue
 import no.nav.familie.pdf.pdf.domain.VerdilisteElement
 
 object TabellUtils {
-    fun lagTabell(
-        tabellData: List<VerdilisteElement>,
-        caption: String,
+    fun visningsVariantTabeller(
+        verdiliste: List<VerdilisteElement>,
+        seksjon: Div,
+    ) {
+        verdiliste.forEach { barn ->
+            barn.verdiliste?.let { seksjon.add(lagTabell(barn)) }
+        }
+    }
+
+    private fun lagTabell(
+        tabellData: VerdilisteElement,
     ): Table {
+        requireNotNull(tabellData.verdiliste) { "VerdilisteElement må ha en verdiliste for å kunne lage en tabell" }
+
         val tabell =
             Table(UnitValue.createPercentArray(floatArrayOf(50f, 50f))).apply {
                 useAllAvailableWidth()
                 setMarginBottom(10f)
                 setMarginLeft(15f)
                 accessibilityProperties.role = StandardRoles.TABLE
             }
+
         val captionDiv =
             Div().apply {
                 add(
-                    Paragraph(caption).apply {
+                    Paragraph(tabellData.label).apply {
                         setFontColor(DeviceRgb(0, 52, 125))
                         setFontSize(14f)
                         simulateBold()
                     },
                 )
             }
         tabell.caption = captionDiv
-
         tabell.addCell(lagTabellOverskriftscelle("Spørsmål"))
         tabell.addCell(lagTabellOverskriftscelle("Svar", false))
-        lagTabellRekursivt(tabellData, tabell)
+        tabell.lagTabellRekursivt(tabellData.verdiliste)
         return tabell
     }
 
-    private fun lagListeMedAlleElementer(
-        elementer: List<VerdilisteElement>,
-        strengManSkalSplitteTabellPå: String,
-    ): List<List<VerdilisteElement>> {
-        val listeMedAlleElementer = mutableListOf<List<VerdilisteElement>>()
-        var nåværendeElement = mutableListOf<VerdilisteElement>()
-        elementer.forEachIndexed { index, item ->
-            if (item.label == strengManSkalSplitteTabellPå && index != 0) {
-                listeMedAlleElementer.add(nåværendeElement)
-                nåværendeElement = mutableListOf()
-            }
-            nåværendeElement.add(item)
+    private fun lagTabellOverskriftscelle(
+        tekst: String,
+        erVenstreKolonne: Boolean = true,
+    ): Cell =
+        Cell().apply {
+            add(
+                Paragraph(tekst).apply {
+                    setFontColor(DeviceRgb(0, 86, 180))
+                    setFontSize(14f)
+                    simulateBold()
+                },
+            )
+            setBorder(Border.NO_BORDER)
+            if (erVenstreKolonne) setPaddingRight(10f) else setPaddingLeft(10f)
+            accessibilityProperties.role = StandardRoles.TH
         }
-        listeMedAlleElementer.add(nåværendeElement)
-        return listeMedAlleElementer
-    }
 
-    private fun lagTabellRekursivt(
+    private fun Table.lagTabellRekursivt(
```

Comment:
Burde vi prøve å være konsekvente på extension-functions vs vanlige funksjoner? Eller er det vanlig å blande så mye som vi gjør?

Reply:
Kan godt ta en felles gjennom gang på det 👍🏼 Jeg kan lage en egen "region" for extension-functions hvis det er det du tenkte på? Ut i fra det jeg har funnet så bruker man extension functions når du vil utvide eksisterende klasse som man selv ikke "eier". https://kotlinlang.org/docs/extensions.html

Hvor mener du vi avviker?